src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -0,0 +1,87 @@
+    // get edits from form
+    String idResponse = request.getParameter("recipeID");
+    String nameResponse = request.getParameter("edit-name");
+    String tagsResponse = request.getParameter("edit-tags");
+    String descriptionResponse = request.getParameter("edit-description");
+    String ingredientsResponse = request.getParameter("edit-ingredients");
+    String stepsResponse = request.getParameter("edit-steps");
+    // privacy will be used once we give users the option to make their recipes public
+    // String privacy = request.getParameter("privacy");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    
+    long id = Long.parseLong(idResponse);
+    Key key = KeyFactory.createKey("PrivateRecipe", id);
+    try {
+      Entity recipeEntity = datastore.get(key);
+
+      // gets saved values from DataStore
+      String name = (String) recipeEntity.getProperty("name");
+      ArrayList<String> tags = (ArrayList<String>) recipeEntity.getProperty("tags");
+      String description = (String) recipeEntity.getProperty("description");
+      ArrayList<String> ingredients = (ArrayList<String>) recipeEntity.getProperty("ingredients");
+      ArrayList<String> steps = (ArrayList<String>) recipeEntity.getProperty("steps");
+
+      // if the name input in the form is not empty, the form value will be saved
+      if (!nameResponse.equals("")) {
+        name = nameResponse;
+        name = name.trim();
+      }
+      // if the tags input in the form is not empty, the form value will be saved
+      if (!tagsResponse.equals("")) {
+        tags = FormHelper.separateByCommas(tagsResponse);
+      }
+      // if the description input in the form is not empty, the form value will be saved
+      if (!descriptionResponse.equals("")) {
+        description = descriptionResponse;
+        description = description.trim();
+      }
+      // if the ingredients input in the form is not empty, the form value will be saved
+      if (!ingredientsResponse.equals("")) {
+        ingredients = FormHelper.separateByNewlines(ingredientsResponse);
+      }
+      // if the steps input in the form is not empty, the form value will be saved
+      if (!stepsResponse.equals("")) {
+        steps = FormHelper.separateByNewlines(stepsResponse);
+      }
+
+      recipeEntity.setProperty("name", name);
+      recipeEntity.setProperty("tags", tags);
+      recipeEntity.setProperty("description", description);
+      recipeEntity.setProperty("ingredients", ingredients);
+      recipeEntity.setProperty("steps", steps);
+      recipeEntity.setProperty("imageBlobKey", (String) recipeEntity.getProperty("imageBlobKey"));
+
+      datastore.put(recipeEntity);
+    } catch (EntityNotFoundException e) {
+      // in normal circumstances, this won't happen bc user has not access to id
+      System.out.println("entity not found exception");
```

Comment:
nit: add a little more detail to this error message so it's obvious where it's coming from / what's happening